src/api/handlers/page/delete_page_handler.rs

```diff
@@ -0,0 +1,61 @@
+use std::sync::Arc;
+use serde::Serialize;
+use crate::{ error::Result };
+use axum::{Extension, extract::Path, Json};
+use axum::extract::State;
+use crate::avored_state::AvoRedState;
+use crate::error::Error;
+use crate::models::token_claim_model::LoggedInUser;
+use crate::models::validation_error::{ErrorMessage, ErrorResponse};
+
+pub async fn delete_page_handler(
+    Extension(logged_in_user): Extension<LoggedInUser>,
+    Path(page_id): Path<String>,
+    state: State<Arc<AvoRedState>>
+) -> Result<Json<RemovedPageResponse>> {
+    let has_permission_bool = state
+        .admin_user_service
+        .has_permission(logged_in_user.clone(), String::from("page_delete"))
+        .await?;
+    if !has_permission_bool {
+        return Err(Error::FORBIDDEN);
+    }
+    let page_result = state
+        .page_service
+        .find_by_id(&state.db, page_id.clone())
+        .await;
+    match page_result {
+        Ok(_) => {
+            state.page_service.remove_by_id(&state.db, &page_id).await?;
+            let response = RemovedPageResponse {
+                status: true
+            };
+            Ok(Json(response))
+        },
+        Err(error) => {
+            match error {
+                Error::Generic(ref message) if message == "no record found" => {
```

Comment:
Over here.. I would suggest just change the `page_repository => find_by_id` with 
 
    let result_object = match result_object_option {
            Some(object) => object,
            None => Err(Error:: NOT_FOUND("no record found".to_string())),
        };

Reply:
Do you mean to replace this code 
```
match page_result {
        Ok(_) => {
            state.page_service.remove_by_id(&state.db, &page_id).await?;
            let response = RemovedPageResponse {
                status: true
            };
            Ok(Json(response))
        },
        Err(error) => {
            match error {
                Error::Generic(ref message) if message == "no record found" => {
                    return Err(Error::NotFound("Page not found".to_string()));
                },
                _ => {
                    let mut errors: Vec<ErrorMessage> = vec![];
                    let error_message = ErrorMessage {
                        key: String::from("email"),
                        message: String::from("bad request error")
                    };
                    errors.push(error_message);
                    let error_response = ErrorResponse {
                        status: false,
                        errors
                    };
                    return Err(Error::BadRequestError(error_response))
                }
            }
        }
    }
```
with this
```
let result_object = match result_object_option {
        Some(object) => object,
        None => Err(Error:: NOT_FOUND("no record found".to_string())),
    };
```
?

Reply:
If you do a change in the page_repository like i mention below in that case over here you can like below. Yes you right but to make that work you have to update the `page_repository.rs=>find_by_id` method too.

     let result_object = match result_object_option {
        Some(object) => object,
        None => Err(Error:: NOT_FOUND("no record found".to_string())),
    };

Reply:
match error {
                Error::NotFound(ref message) => {
                    return Err(Error::NotFound("Page not found".to_string()));
                },
                _ => {
                    let mut errors: Vec<ErrorMessage> = vec![];
                    let error_message = ErrorMessage {
                        key: String::from("email"),
                        message: String::from("bad request error")
                    };
                    errors.push(error_message);
                    let error_response = ErrorResponse {
                        status: false,
                        errors
                    };
                    return Err(Error::BadRequestError(error_response))
                }
            }

Reply:
I have one problem with the remove by id function in the page repository
https://github.com/avored/avored-rust-cms/pull/220/files/b39392ed6eaf90a3efbc006edd8e2bb4a364f062#diff-6b857fc182c441184b7a99443db91411f8ae424b6de7bd6b953dc635b76ad06d

I've refactored due to your suggestion

```
pub async fn remove_by_id(
        &self,
        datastore: &Datastore,
        database_session: &Session,
        page_id: &String) -> Result<bool>
    {
        let sql = format!("DELETE pages:{page_id}");
        let responses = datastore.execute(&sql, database_session, None).await?;
        for item in &responses {
            println!("Item, {:?}", item)
        }
        let result_object_option = into_iter_objects(responses)?.next();
        let result_object = match result_object_option {
            Some(object) => object,
            None => Err(Error::Generic("no record found".to_string())),
        };
        println!("Res, {:?}", result_object);
        Ok(true)
    }
```
The problem is, that I always end up in this case, even if I remove an existing item ```None => Err(Error::Generic("no record found".to_string()))```
This log show always the similar log
```
 for item in &responses {
            println!("Item, {:?}", item)
        }
```
``` Item, Response { time: 5.563125ms, result: Ok(Array(Array([]))), query_type: Other }```
Maybe you have some ideas what am I doing wrong?

Reply:
you can simply do it like below
     
    let responses = datastore.execute(sql, database_session, Some(vars)).await?;

        let response = responses
            .into_iter()
            .next()
            .map(|rp| rp.output());
        let query_result = match response {
            Some(object) => object.is_ok(), // there is another method is_err() as well 
            None => false
        };

As i know surrealdb do not return any result as part of delete query so the response is empty with `Array[[]]`. So basically `is_ok()` method is a response which tells us that query was ok or not.